src/DashboardService.Validation/Priority/EditPriorityRequestValidator.cs

```diff
@@ -0,0 +1,70 @@
+using FluentValidation;
+using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.DashboardService.Validation.Priority.Interfaces;
+using LT.DigitalOffice.Kernel.Validators;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Validation.Priority;
+
+public class EditPriorityRequestValidator : ExtendedEditRequestValidator<Guid, EditPriorityRequest>, IEditPriorityRequestValidator
+{
+  private readonly IPriorityRepository _priorityRepository;
+
+  public EditPriorityRequestValidator(IPriorityRepository priorityRepository)
+  {
+    _priorityRepository = priorityRepository;
+
+    RuleFor(x => x)
+      .CustomAsync(async (x, context, _) =>
+      {
+        foreach (var op in x.Item2.Operations)
+        {
+          await HandleInternalPropertyValidationAsync(op, x.Item1, context);
+        }
+      });
+  }
+
+  private async Task HandleInternalPropertyValidationAsync(
+    Operation<EditPriorityRequest> requestedOperation,
+    Guid priorityId,
+    ValidationContext<(Guid, JsonPatchDocument<EditPriorityRequest>)> context)
+  {
+    RequestedOperation = requestedOperation;
+    Context = context;
+    
+    AddCorrectPaths(
+      new List<string>
+      {
+        nameof(EditPriorityRequest.Name),
+      });
+    
+    AddCorrectOperations(nameof(EditPriorityRequest.Name), new() { OperationType.Replace });
+    
+    AddFailureForPropertyIfNot(
+      nameof(EditPriorityRequest.Name),
+      x => x == OperationType.Replace,
+      new()
+      {
+        { x => !string.IsNullOrEmpty(x.value?.ToString().Trim()), "Name must not be empty." },
+        { x => x.value.ToString().Length < 50, "Name is too long." },
+        
+      }, CascadeMode.Stop);
```

Comment:
doesn't it say about obsolete status? use class level

Reply:
obsolete is StopOnFirstFailure. didn`t understand about class level. what did you mean?